src/script/builder/tir/prim_func_frame.cc

```diff
@@ -22,16 +22,34 @@
 #include <tvm/tir/function.h>
 
 #include "./block_frame.h"
+#include "./var.h"
 
 namespace tvm {
 namespace script {
 namespace builder {
 namespace tir {
 
+void PrimFuncFrameNode::EnterWithScope() {
+  TIRFrameNode::EnterWithScope();
+  // add implicit root block
+  root_block_frame->EnterWithScope();
+}
+
 void PrimFuncFrameNode::ExitWithScope() {
   using namespace tvm::tir;
+  root_block_frame->ExitWithScope();
   TIRFrameNode::ExitWithScope();
   Builder builder = Builder::Current();
+  if (!(stmts.size() == 1 && stmts[0]->IsInstance<BlockRealizeNode>())) {
+    LOG(FATAL) << "ValueError: PrimFuncFrame shoulde have one and only one root block.";
+  }
+  BlockRealize root_block_realize = Downcast<BlockRealize>(stmts[0]);
+  Block root_block = root_block_realize->block;
+  // remove redundant implicit root block
+  if (root_block->alloc_buffers.empty() && root_block->body->IsInstance<BlockRealizeNode>()) {
```

Comment:
the condition is a bit loose here. what about we have other fields defined, like `T.block_attr` and `T.reads/writes`?